src/memo.rs

```diff
@@ -1,16 +1,20 @@
+use crate::cost::Cost;
 use crate::operator::Operator;
+use crate::property::PhysicalProperties;
 use crate::rule::Rule;
 use crate::statistics::Statistics;
 use crate::{LogicalPlan, OptimizerContext, Plan};
 use bit_set::BitSet;
 use std::cell::RefCell;
+use std::collections::HashMap;
 use std::rc::{Rc, Weak};
 
 pub struct GroupPlan {
     group: GroupWeakRef,
     op: Operator,
     inputs: Vec<GroupRef>,
     rule_masks: BitSet,
+    require_to_output_map: HashMap<PhysicalProperties, Rc<PhysicalProperties>>,
```

Comment:
you can `type RequireToOutputMap = HashMap<PhysicalProperties, Rc<PhysicalProperties>>;` and use RequireToOutputMap

Reply:
done